events/src/main/java/com/redhat/runtimes/inventory/events/EventConsumer.java

```diff
@@ -189,6 +156,46 @@ public CompletionStage<Void> processEggFlow(Message<String> message) {
     return message.ack();
   }
 
+  @Transactional
+  public void processMessage(ArchiveAnnouncement announce, String json) {
+    // Needs to be visible in the catch block
+    JvmInstance inst = null;
+    try {
+      InsightsMessage msg = instanceOf(announce, json);
+
+      if (msg instanceof JvmInstance) {
+        inst = (JvmInstance) msg;
+      } else if (msg instanceof EapInstance) {
+        inst = (EapInstance) msg;
+      } else if (msg instanceof UpdateInstance update) {
+        var linkingHash = update.getLinkingHash();
+        var maybeInst = getInstanceFromHash(linkingHash);
+        if (maybeInst.isPresent()) {
+          inst = maybeInst.get();
+          var newJars = update.getUpdates();
+          inst.getJarHashes().addAll(newJars);
+        } else {
+          throw new IllegalStateException(
+              "Update message seen for non-existent hash: " + linkingHash);
+        }
+      } else {
+        // Can't happen, but just in case
+        throw new IllegalStateException("Message seen that is neither a new instance or an update");
+      }
+
+      if (inst != null) {
+        Log.infof("About to persist: %s", inst);
```

Comment:
Maybe this should be logged under debug? When running the new integration test for the egg channel this logs the entire eap instance.

Reply:
+1 to log as debug